xbmc/platform/linux/input/LibInputHandler.cpp

```diff
@@ -0,0 +1,226 @@
+/*
+ *      Copyright (C) 2005-2017 Team XBMC
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "LibInputHandler.h"
+#include "LibInputKeyboard.h"
+#include "LibInputPointer.h"
+#include "LibInputTouch.h"
+
+#include "utils/log.h"
+
+#include <algorithm>
+#include <fcntl.h>
+#include <linux/input.h>
+#include <string.h>
+#include <sys/ioctl.h>
+#include <unistd.h>
+
+static int open_restricted(const char *path, int flags, void __attribute__((unused)) *user_data)
+{
+  int fd = open(path, flags);
+
+  if (fd < 0)
+  {
+    CLog::Log(LOGERROR, "%s - failed to open %s (%s)", __FUNCTION__, path, strerror(errno));
+    return -errno;
+  }
+
+  auto ret = ioctl(fd, EVIOCGRAB, (void*)1);
+  if (ret < 0)
+  {
+    CLog::Log(LOGDEBUG, "%s - grab requested, but failed for %s (%s)", __FUNCTION__, path, strerror(errno));
+  }
+
+  return fd;
+}
+
+static void close_restricted(int fd, void  __attribute__((unused)) *user_data)
+{
+  close(fd);
+}
+
+static const struct libinput_interface m_interface =
+{
+  open_restricted,
+  close_restricted,
+};
+
+static void LogHandler(libinput  __attribute__((unused)) *libinput, libinput_log_priority priority, const char *format, va_list args)
+{
+  if (priority == LIBINPUT_LOG_PRIORITY_DEBUG)
+  {
+    char buf[512];
+    int n = vsnprintf(buf, sizeof(buf), format, args);
+    if (n > 0)
+      CLog::Log(LOGDEBUG, "libinput: %s", buf);
+  }
+}
+
+CLibInputHandler::CLibInputHandler(CWinEventsLinux *winEvents)
+{
+  m_udev = udev_new();
```

Comment:
so, now eudev/systemd is not an optional build time dependency anymore

Reply:
Correct, one or the other

Reply:
there are few issues with this. 

- udev is marked as opional dependency, kodi wont build with -DENABLE_UDEV=OFF -so that should be fixed in cmakefiles now
- libusb is (was) an optional dependency, only if udev was NOT found. see xbmc/peripherals/bus/PeripheralBusUSB.h and xbmc/peripherals/bus/PeripheralBusUSB.*. should this be completely removed now ?

Reply:
this is only true for platforms without "windowing": RPi, AML, GBM
For those udev should be mandatory for those only